packages/fluentui/react-northstar/src/components/Tree/Tree.tsx

```diff
@@ -228,235 +160,41 @@ export const Tree: ComponentWithAs<'div', TreeProps> &
-    if (!selectable || !hasSubtree(item)) {
-      return false;
-    }
-
-    const selectableItemIds = getAllSelectableChildrenId(item.items as TreeItemProps[]);
-
-    return (
-      !isAllGroupChecked(item.items as TreeItemProps[], selectedItemIds) &&
-      selectableItemIds.some(id => selectedItemIds.indexOf(id) > -1)
-    );
-  };
-
-  const isSelectedItem = (item: TreeItemProps): boolean => {
-    if (selectable && hasSubtree(item)) {
-      return isAllGroupChecked(item.items as TreeItemProps[], selectedItemIds);
-    }
-
-    return selectedItemIds && selectedItemIds.indexOf(item.id) > -1;
-  };
+  const {
+    tobeRenderedItemsProps,
+    toggleActive,
+    focusParent,
+    focusFirstChild,
+    siblingsExpand,
+    registerItemRef,
+    toggleSelect,
+    getItemById,
+  } = useSelectableTree(props);
 
   const contextValue: TreeRenderContextValue = React.useMemo(
     () => ({
-      onFocusParent,
-      onSiblingsExpand,
-      onFocusFirstChild,
-      onTitleClick,
+      toggleActive,
+      focusParent,
+      siblingsExpand,
+      focusFirstChild,
+      registerItemRef,
+      toggleSelect,
+      getItemById,
     }),
-    [onFocusParent, onSiblingsExpand, onFocusFirstChild, onTitleClick],
+    [toggleActive, focusParent, siblingsExpand, focusFirstChild, registerItemRef, toggleSelect, getItemById],
   );
 
   const renderContent = (): React.ReactElement[] => {
-    if (!items) return null;
-
-    const renderItems = (items: TreeItemProps[], level = 1, parent?: string): React.ReactElement[] => {
-      return items.reduce((renderedItems: React.ReactElement[], item: TreeItemProps, index: number) => {
-        const id = item.id;
-        const isSubtree = hasSubtree(item);
-        const isSubtreeExpanded = isSubtree && isActiveItem(id);
-
-        const indeterminate = isIndeterminate(item);
-
-        if (!itemsRef.current.has(id)) {
-          itemsRef.current.set(id, React.createRef<HTMLElement>());
-        }
-
-        const renderedItem = TreeItem.create(item, {
-          defaultProps: () =>
-            getA11yProps('item', {
-              expanded: isSubtreeExpanded,
-              selected: isSelectedItem(item),
-              selectable,
-              renderItemTitle,
-              id,
-              key: id,
-              parent,
-              level,
-              index: index + 1, // Used for aria-posinset and it's 1-based.
-              contentRef: itemsRef.current.get(id),
-              treeSize: items.length,
-              indeterminate,
-              onSiblingsExpand,
-            }),
-        });
-
-        return [
-          ...renderedItems,
-          renderedItem,
-          ...(isSubtreeExpanded ? renderItems(item.items as TreeItemProps[], level + 1, id) : ([] as any)),
-        ];
-      }, []);
-    };
-    const itemsRendered = renderItems(items as TreeItemProps[]);
-
-    return itemsRendered;
+    if (!tobeRenderedItemsProps) return [];
```

Comment:
```suggestion
```

Can we always return an array?

Reply:
Oh in fact we are, let me remove this useless check